### PR TITLE
Add section about sort-functions

### PR DESCRIPTION
**Pull request by LinqLover**
See https://github.com/hpi-swa-lab/SqueakByExample-english/issues/51#issuecomment-738720524. I hope this has not become too detailed :D

### CODE REVIEW

File: Collections/Collections.tex [hunk 1]
```diff
@@ -1135,6 +1134,71 @@ \section{Some hints for using collections}
...
+"Out-of-place sort"
+array sorted. --> #(1 3 5)
+array. --> #(3 5 1)
+
+"In-place sort"
+array sort.
+array. --> #(1 3 5)
+\end{code}
+
+Sort methods use the message \mthind{Magnitude}{<=} to establish sort order, so by default, they can sort magnitudes such as numbers, characters, times \etc (see \charef{basic} on \ct{Magnitude}), but also other objects such as strings or points.
```
Comment (codeZeilen) [inline Collections/Collections.tex:1162]: times -> timestamps

File: Collections/Collections.tex [hunk 2]
```diff
@@ -1135,6 +1134,71 @@ \section{Some hints for using collections}
...
+However, what can you do if you want to sort objects that do not implement \ct{<=}, or if you would like to customize sort order?
```
Comment (codeZeilen) [inline Collections/Collections.tex:1163]: "customize sort order" -> "customize the sort order"

File: Collections/Collections.tex [hunk 3]
```diff
@@ -668,15 +668,14 @@ \subsection{Set}
...
-
 You can create a \ct{SortedCollection} by creating a new instance and adding elements to it:
+
 \begin{code}{@TEST}
 SortedCollection new add: 5; add: 2; add: 50; add: -10; yourself. --> a SortedCollection(-10 2 5 50)
 \end{code}
 
-While \ct{SortedCollection}s are quite useful, if the collection should always be in sorted order, they should not be used to simply sort a collection once. You can use the message \mthind{Collection}{sorted}, which returns an array with the sorted elements:
+While \ct{SortedCollection}s are quite useful if the collection should always be in sorted order, they should not be used to simply sort a collection once.
+To achieve the latter, you can use the messages \ct{sort} and \ct{sorted} which are introduced in \secref{sorting}.
```
[the remark this answers is not in the document]
Comment (codeZeilen) [reply Collections/Collections.tex:678]: Or add sorted: to the list

File: Collections/Collections.tex [hunk 4]
```diff
@@ -1135,6 +1134,71 @@ \section{Some hints for using collections}
...
+\end{code}
+
+However, the sort function protocol provides a much more elegant solution to this problem by defining a dedicated \clsind{PropertySortFunction} class.
+It can be instantiated from any unary block by sending the message \mthind{BlockClosure}{ascending} or \mthind{BlockClosure}{descending} to it:
+
+\begin{code}{@TEST}
+(1 to: 10) sorted: [:ea | ea raisedTo: 2 modulo: 3] ascending --> #(3 6 9 1 2 4 5 7 8 10)
+\end{code}
+
+\mthind{Symbol}{ascending} and \mthind{BlockClosure}{descending} are defined on \clsind{Symbol} as well, so we could also write the previous example in the following way (but the former version is more efficient):
```
Comment (codeZeilen) [inline Collections/Collections.tex:1186]: The former version is also considered better style.

**Review by codeZeilen — CHANGES_REQUESTED** [review]

File: SBE.ind [hunk 5]
```diff
@@ -1,1487 +1,1581 @@
 \begin{theindex}
```
[the remark this answers is not in the document]
Comment (codeZeilen) [reply SBE.ind:1]: yeah that should probably be part of the CI o.O But no hurry!

File: Collections/Collections.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (LinqLover) [reply Collections/Collections.tex:678]: As we are talking about sorting a collection once, an in-place sort would be questionary here anyway:

```suggestion
To achieve the latter, you can use the messages \ct{sorted} and \ct{sorted:} which are introduced in \secref{sorting}.
```

File: Collections/Collections.tex (hunk 1 above)
Comment (LinqLover) [reply Collections/Collections.tex:1162]:
```suggestion
Sort methods use the message \mthind{Magnitude}{<=} to establish sort order, so by default, they can sort magnitudes such as numbers, characters, timestamps, \etc (see \charef{basic} on \ct{Magnitude}), but also other objects such as strings or points.
```

But `Time` inherits from `Magnitude` as well :-)

File: Collections/Collections.tex (hunk 2 above)
Comment (LinqLover) [reply Collections/Collections.tex:1163]:
```suggestion
However, what can you do if you want to sort objects that do not implement \ct{<=}, or if you would like to customize the sort order?
```

File: Collections/Collections.tex (hunk 4 above)
Comment (LinqLover) [reply Collections/Collections.tex:1186]:
```suggestion
\mthind{Symbol}{ascending} and \mthind{BlockClosure}{descending} are defined on \clsind{Symbol} as well, so we could also write the previous example in the following way (however, the former version is more efficient, and it also would be considered better style):
```

**Comment by LinqLover** [comment]
@codeZeilen Please feel free to accept the pending suggestions if you like them :-)

**Review by codeZeilen — APPROVED** [review]

File: SBE.ind (hunk 5 above)
[the remark this answers is not in the document]
Comment (LinqLover) [reply SBE.ind:1]: Opened #91

**Pull request merged**